uvicorn/_types.py

```diff
@@ -0,0 +1,87 @@
+import sys
+from typing import Optional, Union
+
+if sys.version_info < (3, 8):
+    from typing_extensions import Literal, TypedDict
+else:
+    from typing import Literal, TypedDict
+
+# SCOPES
+
+
+class ASGIDict(TypedDict):
```

Comment:
Maybe this alternative naming…? (We know it's a `dict`, so we might want to indeed answer "what about ASGI does it contain?".)

```suggestion
class ASGISpecInfo(TypedDict):
```

Reply:
done yes